xbmc/addons/interfaces/GUI/ListItem.cpp

```diff
@@ -87,5 +112,236 @@ void Interface_GUIListItem::destroy(void* kodiBase, void* handle)
+                          __FUNCTION__, addon, item, label, addon ? addon->ID().c_str() : "unknown");
+    return;
+  }
+
+  if (item->get() == nullptr)
+  {
+    CLog::Log(LOGERROR, "Interface_GUIListItem::%s - empty list item called on addon '%s'", __FUNCTION__, addon->ID().c_str());
+    return;
+  }
+
+  Interface_GUIGeneral::lock();
+  item->get()->SetLabel(label);
+  Interface_GUIGeneral::unlock();
+}
+
+char* Interface_GUIListItem::get_label2(void* kodiBase, void* handle)
+{
+  CAddonDll* addon = static_cast<CAddonDll*>(kodiBase);
+  CFileItemPtr* item = static_cast<CFileItemPtr*>(handle);
+  if (!addon || !item)
+  {
+    CLog::Log(LOGERROR, "Interface_GUIListItem::%s - invalid handler data (kodiBase='%p', handle='%p') on addon '%s'",
+                          __FUNCTION__, addon, item, addon ? addon->ID().c_str() : "unknown");
+    return nullptr;
+  }
+
+  if (item->get() == nullptr)
+  {
+    CLog::Log(LOGERROR, "Interface_GUIListItem::%s - empty list item called on addon '%s'", __FUNCTION__, addon->ID().c_str());
+    return nullptr;
+  }
+
+  char* ret;
+  Interface_GUIGeneral::lock();
+  ret = strdup(item->get()->GetLabel2().c_str());
+  Interface_GUIGeneral::unlock();
+  return ret;
+}
+
+void Interface_GUIListItem::set_label2(void* kodiBase, void* handle, const char *label)
+{
+  CAddonDll* addon = static_cast<CAddonDll*>(kodiBase);
+  CFileItemPtr* item = static_cast<CFileItemPtr*>(handle);
+  if (!addon || !item || !label)
+  {
+    CLog::Log(LOGERROR, "Interface_GUIListItem::%s - invalid handler data (kodiBase='%p', handle='%p', label='%p') on addon '%s'",
+                          __FUNCTION__, addon, item, label, addon ? addon->ID().c_str() : "unknown");
+    return;
+  }
+
+  if (item->get() == nullptr)
+  {
+    CLog::Log(LOGERROR, "Interface_GUIListItem::%s - empty list item called on addon '%s'", __FUNCTION__, addon->ID().c_str());
+    return;
+  }
+
+  Interface_GUIGeneral::lock();
+  item->get()->SetLabel2(label);
+  Interface_GUIGeneral::unlock();
+}
+
+char* Interface_GUIListItem::get_icon_image(void* kodiBase, void* handle)
+{
+  CAddonDll* addon = static_cast<CAddonDll*>(kodiBase);
+  CFileItemPtr* item = static_cast<CFileItemPtr*>(handle);
+  if (!addon || !item)
+  {
+    CLog::Log(LOGERROR, "Interface_GUIListItem::%s - invalid handler data (kodiBase='%p', handle='%p') on addon '%s'",
+                          __FUNCTION__, addon, item, addon ? addon->ID().c_str() : "unknown");
+    return nullptr;
+  }
+
+  if (item->get() == nullptr)
+  {
+    CLog::Log(LOGERROR, "Interface_GUIListItem::%s - empty list item called on addon '%s'", __FUNCTION__, addon->ID().c_str());
+    return nullptr;
+  }
+
+  char* ret;
+  Interface_GUIGeneral::lock();
+  ret = strdup(item->get()->GetIconImage().c_str());
+  Interface_GUIGeneral::unlock();
+  return ret;
+}
+
+void Interface_GUIListItem::set_icon_image(void* kodiBase, void* handle, const char *label)
```

Comment:
Copy/paste error? Parameter name <code>label</code> does not match what it actually contains. Should read 'image' or the like. 

I've seen this name mismatch in more setters in this source file. All names should be fixed.